pandas/core/generic.py

```diff
@@ -8029,6 +8029,91 @@ def mask(self, cond, other=np.nan, inplace=False, axis=None, level=None,
         is not realigned. That is, use freq if you would like to extend the
         index when shifting and preserve the original data.
 
+        Examples
+        --------
+
+        >>> df = pd.DataFrame({'group': ['A', 'A', 'A', 'B', 'B', 'B'],
+        ...                    'myvalue': [1, 2, 3, 4, 5, 6]},
+        ...                     index=pd.DatetimeIndex(['2016-06-06',
+        ...                                             '2016-06-08',
+        ...                                             '2016-06-09',
+        ...                                             '2016-06-10',
+        ...                                             '2016-06-12',
+        ...                                             '2016-06-13'],
+        ...                                            name='mydate'))
+
+        >>> df
+                    group	myvalue
+        mydate
+        2016-06-06	A	1
+        2016-06-08	A	2
+        2016-06-09	A	3
+        2016-06-10	B	4
+        2016-06-12	B	5
+        2016-06-13	B	6
+
+        For the groups compute the difference between current `myvalue` and
+        `myvalue` shifted forward by 1 day.
+
+        If `myvalue` is shifted then the values will simply move down.
+
+        >>> df.myvalue.shift(1)
+        mydate
+        2016-06-06    NaN
+        2016-06-08    1.0
+        2016-06-09    2.0
+        2016-06-10    3.0
+        2016-06-12    4.0
+        2016-06-13    5.0
+        Name: myvalue, dtype: float64
+
+        We only want to shift `myvalue` forward by one day before computing
```

Comment:
Just for clearer wording I'd say "If instead you wanted to shift values forward by a day you can do this by reindexing first and filling `key`"